### PR TITLE
✨ feat: UNT-T24589: Create signup and signin

### PR DESCRIPTION
**Pull request by ami-b-simform**
Design: 
image

Actual implementation:
Signin:
simulator_screenshot_7C715B29-7063-420F-94BD-1444B7ABDABF

Signup:
simulator_screenshot_CC3AB568-1E17-437E-BA20-54F6C85C76F8

### CODE REVIEW

File: lib/modules/auth/signin.dart [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+import 'package:podcasty/widgets/custom_button.dart';
+import 'package:podcasty/widgets/input_field.dart';
+
+import '../../values/app_colors.dart';
+
+class SignInScreen extends StatefulWidget {
+  const SignInScreen({Key? key}) : super(key: key);
+
+  @override
+  SigninForm createState() => SigninForm();
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: The state class should be private and please check the naming convention as well.

File: lib/modules/auth/signin.dart [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+import 'package:flutter/material.dart';
+import 'package:podcasty/modules/auth/signup.dart';
+import 'package:podcasty/modules/auth/widgets/form_header.dart';
+import 'package:podcasty/modules/auth/widgets/header_widget.dart';
+import 'package:podcasty/values/app_theme.dart';
+import 'package:podcasty/values/string_constants.dart';
+import 'package:podcasty/widgets/custom_button.dart';
+import 'package:podcasty/widgets/input_field.dart';
+
+import '../../values/app_colors.dart';
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please maintain consistency for the style of import.

File: lib/modules/auth/signin.dart [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+class SignInScreen extends StatefulWidget {
+  const SignInScreen({Key? key}) : super(key: key);
+
+  @override
+  SigninForm createState() => SigninForm();
+}
+
+class SigninForm extends State<SignInScreen> {
+  final signinFormKey = GlobalKey<FormState>();
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please dispose this as well.

File: lib/modules/auth/signin.dart [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+  bool isObscure = true;
+  bool rememberMe = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        body: SafeArea(
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please check format of this file.

File: lib/modules/auth/signin.dart [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+      bottom: false,
+      child: SingleChildScrollView(
+        physics: const BouncingScrollPhysics(),
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please avoid changing physics unless it is absolutely required.

File: lib/modules/auth/signin.dart [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+                        padding: EdgeInsets.symmetric(
+                            horizontal:
+                                MediaQuery.of(context).size.width * 0.05),
+                        child: CustomButton(
+                          text: AppStrings.signIn,
+                          onPressed: () {
+                            if (signinFormKey.currentState!.validate()) {
+                              ScaffoldMessenger.of(context).showSnackBar(
+                                const SnackBar(
+                                  content: Text('Hey! There '),
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please move to constants.

File: lib/modules/auth/signin.dart [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+                      SizedBox(
+                          height: MediaQuery.of(context).size.height * 0.08),
+                      Padding(
+                        padding: EdgeInsets.symmetric(
+                            horizontal:
+                                MediaQuery.of(context).size.width * 0.05),
+                        child: CustomButton(
+                          text: AppStrings.signIn,
+                          onPressed: () {
+                            if (signinFormKey.currentState!.validate()) {
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please avoid force unwrap at all times.

File: lib/modules/auth/signin.dart [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+              height: MediaQuery.of(context).size.height * 0.8 + 16,
+              decoration: const BoxDecoration(
+                color: AppColors.blueGrey,
+                borderRadius: BorderRadius.only(
+                  topLeft: Radius.circular(130.0),
+                ),
+              ),
+              child: Padding(
+                padding: EdgeInsets.symmetric(
+                    horizontal: MediaQuery.of(context).size.width * 0.04),
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please avoid repeated calls for inherited widget.

File: lib/modules/auth/signin.dart [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+                    crossAxisAlignment: CrossAxisAlignment.stretch,
+                    children: [
+                      SizedBox(
+                          height: MediaQuery.of(context).size.height * 0.05),
+                      const FormHeader(
+                        headerText: AppStrings.signIn,
+                        isLeftOriented: false,
+                      ),
+                      SizedBox(
+                          height: MediaQuery.of(context).size.height * 0.05),
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please avoid performing same computation multiple times.

File: lib/modules/auth/signin.dart [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+                        hintText: AppStrings.hintPassword,
+                        validationType: ValidationType.password,
+                        textInputAction: TextInputAction.done,
+                        keyboardType: TextInputType.visiblePassword,
+                        suffixIcon: isObscure
+                            ? Icons.remove_red_eye_outlined
+                            : Icons.remove_red_eye,
+                        obscureText: isObscure,
+                        suffixIconPress: () {
+                          setState(() {
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please use expression body here.

File: lib/modules/auth/signin.dart [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+                            horizontal:
+                                MediaQuery.of(context).size.width * 0.05),
+                        child: Row(
+                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                          children: [
+                            Row(
+                              children: <Widget>[
+                                Checkbox(
+                                  shape: RoundedRectangleBorder(
+                                    borderRadius: BorderRadius.circular(2.0),
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please switch to const version of this.

File: lib/modules/auth/signin.dart [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+                                  ),
+                                  side: MaterialStateBorderSide.resolveWith(
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: You can use all here I believe.

File: lib/modules/auth/signin.dart [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+                                    (states) => const BorderSide(
+                                        width: 1.0,
+                                        color: AppColors.lightYellow),
+                                  ),
+                                  value: rememberMe,
+                                  activeColor: Colors.transparent,
+                                  onChanged: (bool? value) {
+                                    setState(() {
+                                      rememberMe = value!;
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please avoid force unwrap without checking for null value.

File: lib/modules/auth/signin.dart [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+                                ),
+                                const Text(
+                                  AppStrings.rememberMe,
+                                  style: AppTheme.displayMedium,
+                                ),
+                              ],
+                            ),
+                            InkWell(
+                              onTap: () {
+                                ScaffoldMessenger.of(context).showSnackBar(
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: You should create a common way of accessing this.

File: lib/modules/auth/signin.dart [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+                                  },
+                                ),
+                                const Text(
+                                  AppStrings.rememberMe,
+                                  style: AppTheme.displayMedium,
+                                ),
+                              ],
+                            ),
+                            InkWell(
+                              onTap: () {
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: Please extract this function as its body is of more than 3 lines.

File: lib/modules/auth/signin.dart [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+                                  Navigator.push(
+                                    context,
+                                    CupertinoPageRoute(
+                                      builder: (context) =>
+                                          const SignUpScreen(),
+                                    ),
+                                  );
+                                },
+                                child: const Text(
+                                  ' ${AppStrings.register}',
```
Comment (aditya-css) [inline lib/modules/auth/signin.dart]: You can utilise sprintf here.

File: lib/modules/auth/signup.dart [hunk 17]
```diff
@@ -0,0 +1,237 @@
...
+class SignupForm extends State<SignUpScreen> {
+  final signupFormKey = GlobalKey<FormState>();
+  bool isObscure = true;
+  bool rememberMe = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        backgroundColor: AppColors.white,
+        body: Container(
```
Comment (aditya-css) [inline lib/modules/auth/signup.dart]: This seems unnecessary here.

**Review by aditya-css — CHANGES_REQUESTED** [review]
Please follow DRY principle.

File: lib/modules/auth/signin.dart (hunk 12 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 2 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 1 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 3 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 5 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 4 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 6 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 8 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 10 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 9 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 7 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 11 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 13 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signup.dart (hunk 17 above)
Comment (ami-b-simform) [reply lib/modules/auth/signup.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 14 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/modules/auth/signin.dart (hunk 15 above)
Comment (ami-b-simform) [reply lib/modules/auth/signin.dart]: Resolved

File: lib/main.dart [hunk 18]
```diff
@@ -8,14 +10,8 @@ class Podcasty extends StatelessWidget {
...
-        appBar: AppBar(
-          title: Text('Podcasty'),
-        ),
-        body: Container(),
-        // This container represents the blank screen
-      ),
-    );
+        debugShowCheckedModeBanner: false,
+        theme: AppTheme.appThemeLight,
+        home: const SignInScreen());
```
Comment (aditya-css) [inline lib/main.dart]: Please add comma.

File: lib/modules/auth/signin.dart
Comment (aditya-css) [inline lib/modules/auth/signin.dart:1]: Please check this file.

File: lib/modules/auth/utils/snack_bar_utils.dart [hunk 19]
```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/material.dart';
+import 'package:podcasty/values/app_colors.dart';
+
+class SnackBarUtils {
+  static void showSnackBar(BuildContext context, String message) {
```
Comment (aditya-css) [inline lib/modules/auth/utils/snack_bar_utils.dart]: Please don't pass context like this.

File: lib/modules/auth/utils/snack_bar_utils.dart [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+    ScaffoldMessenger.of(context).showSnackBar(
+      SnackBar(
+        content: Text(message,
+            style: const TextStyle(
+              color: AppColors.black,
+              fontSize: 20.0,
+            )),
```
Comment (aditya-css) [inline lib/modules/auth/utils/snack_bar_utils.dart]: Please add comma.

File: lib/modules/auth/widgets/auth_form.dart [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+  final formKey = GlobalKey<FormState>();
+  bool isObscure = true;
+  bool rememberMe = false;
+
+  @override
+  Widget build(BuildContext context) {
+    Size screenSize = MediaQuery.of(context).size;
+    double height = screenSize.height * 0.05;
+    double width = screenSize.width * 0.05;
+    double smallHeight = screenSize.height * 0.02;
```
Comment (aditya-css) [inline lib/modules/auth/widgets/auth_form.dart]: These variables should be final.

Comment (aditya-css) [reply lib/modules/auth/widgets/auth_form.dart]: You don't have to specify data type for local variables.

File: lib/modules/auth/widgets/auth_form.dart [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        HeaderWidget(
+          isLeftOriented: widget.isSignUp ? true : false,
```
Comment (aditya-css) [inline lib/modules/auth/widgets/auth_form.dart]: Ternary seems unnecessary here.

File: lib/modules/auth/widgets/auth_form.dart [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+                  SizedBox(height: screenSize.height * 0.08),
+                  Padding(
+                    padding: EdgeInsets.symmetric(horizontal: width),
+                    child: CustomButton(
+                      appearance: widget.appearance,
+                      text: widget.buttonText,
+                      onPressed: () {
+                        if (formKey.currentState?.validate() ?? false) {
+                          widget.onPressed();
+                          widget.onPressed();
```
Comment (aditya-css) [inline lib/modules/auth/widgets/auth_form.dart]: Why is it called twice?

File: lib/modules/auth/widgets/auth_form.dart [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+                                    Radius.circular(2.0),
+                                  ),
+                                ),
+                                side: const BorderSide(
+                                  width: 1.0,
+                                  color: AppColors.lightYellow,
+                                ),
+                                value: rememberMe,
+                                activeColor: Colors.transparent,
+                                onChanged: (bool? value) {
```
Comment (aditya-css) [inline lib/modules/auth/widgets/auth_form.dart]: Function having body of more than 3 lines should be extracted.

File: lib/modules/auth/widgets/auth_form.dart [hunk 25]
```diff
@@ -0,0 +1,184 @@
...
+                                  borderRadius: BorderRadius.all(
+                                    Radius.circular(2.0),
+                                  ),
+                                ),
+                                side: const BorderSide(
+                                  width: 1.0,
+                                  color: AppColors.lightYellow,
+                                ),
+                                value: rememberMe,
+                                activeColor: Colors.transparent,
```
Comment (aditya-css) [inline lib/modules/auth/widgets/auth_form.dart]: Colors should come from color constants only.

File: lib/modules/auth/widgets/auth_form.dart [hunk 26]
```diff
@@ -0,0 +1,184 @@
...
+                    ),
+                  ),
+                  SizedBox(height: screenSize.height * 0.03),
+                  Padding(
+                    padding: EdgeInsets.symmetric(horizontal: width),
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.end,
+                      children: [
+                        Text(
+                          widget.footerText,
```
Comment (aditya-css) [inline lib/modules/auth/widgets/auth_form.dart:155]: Please handle maxline and other related properties for dynamic text.

File: lib/modules/auth/widgets/auth_form.dart [hunk 27]
```diff
@@ -0,0 +1,184 @@
...
+      ],
+    );
+  }
+
+  @override
+  void dispose() {
+    formKey.currentState?.dispose();
+    super.dispose();
+  }
+}
```
Comment (aditya-css) [inline lib/modules/auth/widgets/auth_form.dart]: This is unnecessary.

File: lib/values/app_theme.dart [hunk 28]
```diff
@@ -9,10 +9,17 @@ class AppTheme {
...
+    fontFamily: ubuntuFont,
+  );
+
   //To display medium fonts
   static const TextStyle displayMedium = TextStyle(
     color: AppColors.lightYellow,
-    fontSize: 16,
+    fontSize: 14,
     fontFamily: ubuntuFont,
   );
```
Comment (aditya-css) [inline lib/values/app_theme.dart]: These should be in text theme only.

File: lib/values/string_constants.dart [hunk 29]
```diff
@@ -18,6 +18,7 @@ class AppStrings {
   static const String dontHaveAccount = "Don't have an account?";
   static const String register = 'Register now';
   static const String forgotPasswordValue = 'Toh rehndo phir 😜';
+  static const String signedIn = 'Hey There! ';
```
Comment (aditya-css) [inline lib/values/string_constants.dart]: There shouldn't be whitespace like this in string.

**Review by aditya-css — CHANGES_REQUESTED** [review]

**Review by aditya-css — APPROVED** [review]

**Pull request merged**